BEASTIE/extractHets.py

```diff
@@ -25,55 +39,98 @@ def count_all_het_sites(tmp,sample,vcfFilename,file_dir,outputFilename,chr_start
             logging.info('..... Working on chr {0} with {1} genes'.format(Num,len(geneList)))
             byGene={}
-            byGene_trans={}
             total_biSNP = 0
+
+            region_str_to_transcripts = {}
             for gene in geneList:
                 Num_transcript = gene.getNumTranscripts()
                 for n in range(Num_transcript):
                     transcript=gene.getIthTranscript(n)
-                    transID = transcript.getTranscriptId()# column 6
                     geneID=transcript.getGeneId()      # column 5
                     byGene[geneID]=byGene.get(geneID, set())
-                    byGene_trans[geneID]=byGene_trans.get(geneID, set())
                     chrom=transcript.getSubstrate()      # column 1
                     chromN=chrom.strip("chr")
                     rawExons=transcript.getRawExons()
                     for exon in rawExons:
                         begin=exon.getBegin()    # column 7
                         end=exon.getEnd()      # column 8
-                        cmd = "tabix " + vcfFilename + " "+chromN+":"+str(begin)+"-"+str(end)#
-                        #tabix /data/allenlab/scarlett/data/VCF/GSD/DNA_vcf/125249.vcf.recode.vcf.gz 1:10042358-10045556
-                        output=Pipe.run(cmd)
-                        if(len(output)==0):continue
-                        lines=output.split("\n")
-                        for line in lines:
-                            fields=line.split("\t")
-                            if(fields[6]!="PASS"): continue
-                            pos=fields[1]                         # column 9
-                            rs = fields[2]                        # column 10
-                            genotype = fields[9].split(':')[0]
-                            if(not isHeterozygous(str(genotype))):continue # go back to the begining of the loop
-                            transcriptCoord=transcript.mapToTranscript(int(pos))#
-                            total_biSNP += 1
-                            chr_pos=chromN+"_"+pos
-                            byGene[geneID].add(chr_pos)
-                            out_stream.write("\t".join([str(chrom),str(chromN),str(geneID),str(pos),str(transID),str(transcriptCoord),str(rs),str(genotype),"\n"]))
-            # write up the basic information
-            # all_set_list = []
-            # non_empty_count = 0
-            # non_empty_count_len = 0
-            # for each_key in byGene:
-            #     all_set_list.extend(list(byGene[each_key]))
-            #     if len(byGene[each_key])!= 0:
-            #         non_empty_count += 1
-            #         non_empty_count_len += len(byGene[each_key])
+
+                        region_str = f"{chromN}:{begin}-{end}"
+                        if not region_str in region_str_to_transcripts:
+                            region_str_to_transcripts[region_str] = []
+                        # else:
```

Comment:
Can you delete the commented out code here?

Reply:
This code is here as it helped expose a difference in functionality between the optimized and unoptimized versions.  @x811zou  has said that the difference is actually an improvement but for this initial diff I wanted to retain the exact same behavior while improving performance.

Reply:
Okay, that's fine. Can you write that in a comment in the code? I think it's good practice to explain why commented-out code hasn't just been deleted alongside said code.